plugins/destination/file/client/json.go

```diff
@@ -0,0 +1,62 @@
+package client
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+const maxJsonSize = 1024 * 1024 * 20
+
+func (c *Client) writeJSONResource(ctx context.Context, tableName string, resources <-chan []interface{}) error {
+	f, err := c.openAppendOnly(ctx, tableName+".json")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	for r := range resources {
+		b, err := json.Marshal(r)
```

Comment:
In the future a drop in encoding/json replacement (eg. `fastjson`) can be used here.